File: array-hash/242-Valid-Anagram.go
func isAnagram(s string, t string) bool {
	if len(s) != len(t) {
		return false
	}
	character_map := make(map[rune]int)

	for _, char := range s {
		character_map[char]++
	}

	for _, char := range t {
		character_map[char]--
	}

	for _, value := range character_map {
		if value > 0 {
			return false
		}
	}

	return true
}

File: reference/array-hash/242-Valid-Anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False

        character_map = {}

        for char in s:
            character_map[char] = character_map.get(char, 0) + 1

        for char in t:
            character_map[char] = character_map.get(char, 0) - 1

        return all(value == 0 for value in character_map.values())